data-prepper-plugins/peer-forwarder/src/main/java/com/amazon/dataprepper/plugins/processor/peerforwarder/discovery/PeerListProviderFactory.java

```diff
@@ -1,24 +1,39 @@
 package com.amazon.dataprepper.plugins.processor.peerforwarder.discovery;
 
 import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.linecorp.armeria.client.endpoint.dns.DnsAddressEndpointGroup;
 
 import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
 
-import static com.amazon.dataprepper.plugins.processor.peerforwarder.PeerForwarderConfig.DATA_PREPPER_IPS;
+import static com.amazon.dataprepper.plugins.processor.peerforwarder.PeerForwarderConfig.HOSTNAME_FOR_DNS_LOOKUP;
+import static com.amazon.dataprepper.plugins.processor.peerforwarder.PeerForwarderConfig.STATIC_ENDPOINTS;
 import static com.amazon.dataprepper.plugins.processor.peerforwarder.PeerForwarderConfig.DISCOVERY_MODE;
 
 public class PeerListProviderFactory {
+    // TODO: Make these configurable?
+    private static final int MIN_TTL = 10;
+    private static final int MAX_TTL = 20;
+
     public PeerListProvider createProvider(final PluginSetting pluginSetting) {
         Objects.requireNonNull(pluginSetting);
 
         final String discoveryModeString = pluginSetting.getStringOrDefault(DISCOVERY_MODE, DiscoveryMode.STATIC.toString()).toUpperCase();
         final DiscoveryMode discoveryMode = DiscoveryMode.valueOf(discoveryModeString);
 
         switch (discoveryMode) {
+            case DNS:
+                final String hostname = pluginSetting.getStringOrDefault(HOSTNAME_FOR_DNS_LOOKUP, null);
+                Objects.requireNonNull(hostname);
+
+                final DnsAddressEndpointGroup endpointGroup = DnsAddressEndpointGroup.builder(hostname)
+                        .ttl(MIN_TTL, MAX_TTL)
+                        .build();
+
+                return new DnsPeerListProvider(endpointGroup);
             case STATIC:
-                final List<String> endpoints = (List<String>) pluginSetting.getAttributeOrDefault(DATA_PREPPER_IPS, Collections.emptyList());
+                final List<String> endpoints = (List<String>) pluginSetting.getAttributeOrDefault(STATIC_ENDPOINTS, Collections.emptyList());
```

Comment:
Should we require that this list isn't empty (similar to the Objects.requireNonNull call in the DNS case)?

Reply:
Good question - originally I thought to automatically default to a single "localhost" endpoint list (while logging a WARN) if an actual list isn't provided, but on second thought this behavior probably does more harm than good. Will update to blow up if a list isn't provided.